contracts/integrations/test/actors/fee_recipient.ts

```diff
@@ -0,0 +1,47 @@
+import { BaseContract } from '@0x/base-contract';
+import { ApprovalFactory, SignedCoordinatorApproval } from '@0x/contracts-coordinator';
+import { SignatureType, SignedZeroExTransaction } from '@0x/types';
+
+import { Actor, ActorConfig, Constructor } from './base';
+
+export interface FeeRecipientConfig extends ActorConfig {
+    verifyingContract?: BaseContract;
+}
+
+export function FeeRecipientMixin<TBase extends Constructor>(Base: TBase) {
```

Comment:
Why is this a mixin/decorator? Seems to impose very specific requirements on constructor args *and* the `TBase` type (which should be `Actor`?). This tight coupling smells like a decorator is the wrong pattern here. I kind of feel like this should just be a simple derived class of `Actor` like:

```ts
export class FeeRecipient extends Actor {
    constructor(cfg: FeeRecipientConfig) {
      super(cfg);
      ...
   }
}
````

Reply:
idk if you've seen the mixins in #2273 but the motivation is to be able to mix e.g. `Maker` and `PoolOperator` together. `FeeRecipient` will probably be less commonly used than some of the others, but one could imagine a test in which you'd want a maker that also signs Coordinator fee recipient approvals. We _could_ make FeeRecipient just derive from `Actor`, but I made it a mixin for consistency